### PR TITLE
Refactor/2318/migrate artificial intelligence component

### LINKED ISSUE #3031: Suspicious metrics badge is only shown initially
# Bug

## Expected Behavior
Given the badge for suspicious metrics isn't shown
When the suspicious metrics change (e.g. due to a map load or black listening something
Then the badge is shown again

## Actual Behavior
The badge is only shown once initially and gets hidden after one view of suspicious metrics. Afterwards it is never shown again.

## Screenshots
image

## Specifications
- is released in [online-demo](https://maibornwolff.github.io/codecharta/visualization/app/index.html?file=codecharta.cc.json.gz&file=codecharta_analysis.cc.json.gz): yes
- CodeCharta Version: 1.105.0

### PR DESCRIPTION
**Pull request by shaman-apprentice**
fix #3031 
ref #2318

### CODE REVIEW

**Review by shaman-apprentice — COMMENTED** [review]
Todo:
- [ ] add changelog

@ce-bo artificialIntelligence.component was basically implemented by you, wasn't it? Do you have time to for the review? I mainly split the two buttons into two components and did a little css adjustment due to Angular components.

File: visualization/app/codeCharta/ui/ribbonBar/artificialIntelligence/suspiciousMetrics/suspiciousMetrics.component.html [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<button [matMenuTriggerFor]="menu" title="Open Suspicious Metrics Panel" class="cc-ribbon-bar-menu-button" (click)="hideBadge = true">
+	<i class="fa fa-exclamation-triangle"></i>
+	<span
+		class="suspicious-metrics-badge"
+		*ngIf="
+			!hideBadge &&
```
[the remark this answers is not in the document]
Comment (ce-bo) [reply visualization/app/codeCharta/ui/ribbonBar/artificialIntelligence/suspiciousMetrics/suspiciousMetrics.component.html:6]: I thought this is already done (at least for loading a new map). We can do this also on blacklisting files.

**Review by ce-bo — APPROVED** [review]
LGTM! One minor remark: We could add some more spaces in between the three categories (suspicious, untracked, unsuspicious metrics). Furthermore, would you please check, why the red badge is not appearing after loading a new map? Thank you!

File: visualization/app/codeCharta/ui/ribbonBar/artificialIntelligence/suspiciousMetrics/suspiciousMetrics.component.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (shaman-apprentice) [reply visualization/app/codeCharta/ui/ribbonBar/artificialIntelligence/suspiciousMetrics/suspiciousMetrics.component.html:6]: this isn't done so far. But I will create a FR and reshow it on every change of metric data (new map, blacklist items, ect.) And I like a changelog entry with a feature way more than just migrating :)

[the remark this answers is not in the document]
Comment (shaman-apprentice) [reply visualization/app/codeCharta/ui/ribbonBar/artificialIntelligence/suspiciousMetrics/suspiciousMetrics.component.html:6]: see #3031

**Pull request merged**